lib/protobuf/message/fields.rb

```diff
@@ -92,20 +94,42 @@ def get_field(name_or_tag, allow_extension = false)
           end
         end
 
+        def extension_shortcuts
+          @extension_shortcuts ||= Set.new
+        end
+
         def define_field(rule, type_class, field_name, tag, options)
           raise_if_tag_collision(tag, field_name)
           raise_if_name_collision(field_name)
 
-          field = ::Protobuf::Field.build(self, rule, type_class, field_name, tag, options)
+          if options[:extension]
```

Comment:
Perhaps worth typing out a sample string of what is what throughout this? Little confusing to read.